mopidy/core/playback.py

```diff
@@ -251,6 +251,10 @@ def _on_about_to_finish(self):
         if self._state == PlaybackState.STOPPED:
             return
 
+        # Unless overridden by other calls (e.g. next / previous / stop) this
+        # will be the last position recorded until the track gets reassigned.
+        self._last_position = self._current_tl_track.track.length
```

Comment:
If memory serves we can set track length to `None` to signify a an "unlimited" length such as web streams. So thinking over this again a "bad" backend that doesn't set a length for a track will just look like a stream. And streaming has been working just fine so I don't think there is much we can or should do about this after all.